docs/proposals/kubeedgeadm-scope.md

```diff
@@ -0,0 +1,122 @@
+Hence proposing the following commands for KubeEdge installation process.
+
+For cloud, commands shall be:
+
+- `kubeedge cloud init`
+- `kubeedge cloud reset`
+
+For edge, commands shall be:
+
+- `kubeedge node init`
+- `kubeedge node reset`
+- `kubeedge node join`
+
+**NOTE:**
+`node` key is used for edge component in the command, for superficial reasons. Because `kubeedge edge init` had `edge` used twice and didn't sound nice.
+
+**ASSUMPTIONS**
+1. GO will **NOT** be installed with any of the commands or processes below. User herself/himself needs to do the same in the target environment. Also it is understood and assumed that user has already built edge-controller and edge_core using the build steps mentioned in README.md
+2. Any failure report by `kubeadm init` or `kubelet`, it is possible that there may be issues with the environment. KubeEdge documentations will address **ONLY** its bring up issues.
+
+## Scope of commands
+
+**Cloud commands:**
+
+`kubeedge cloud init`
+  - What is it?
+     * This command will be responsible to bring up KubeEdge cloud components like edge-controller and K8S (using kubeadm)
+   
+  - What shall be its scope ?
+    1. Check version of OS and install subsequently the required pre-requisites using supported steps. Currently we will support **ONLY** (Ubuntu & CentOS)
+    2. Check and install all the pre-requisites before executing edge-controller, which are
+        * docker (currently 18.06.0ce3-0~ubuntu) and check if service is up
+        * kubelet, kubeadm & kubectl (latest version)
+        * openssl (latest available in OS repos)
+    3. Generate certificates using openssl and save the certs in a predefined static path.
+    It will also compress the folder and display on the terminal so that user can pick it up and transfer it to edge node (VM/host) manually.
+    4. It will update the certificate information in `controller.yaml`
+    5. Start `kubeadm init`.
+
+       **NOTE:** If any issues or error reported from `kubeadm init`, `kubeedge cloud init` command shall be not responsible, as it may occur due to the environment. User has to resolve it. We are taking up this activity to have 1 click install approach for KubeEdge and not K8S.
+
+    6. Update `/etc/kubernetes/manifests/kube-apiserver.yaml` with below information
+    ```
+    - --insecure-port=8080
+    - --insecure-bind-address=0.0.0.0
+    ```
+
+    7. start edge-controller
+
+`kubeedge cloud reset`
+  - What is it? 
+    * This command will be responsible to bring down KubeEdge cloud components edge-controller and call `kubeadm reset` (to stop K8S)
+
+  - What shall be its scope ?
+    1. It shall get the nodes and execute `kubectl drain --force`.
+    2. Kill `edge-controller` process
+    3. Execute `kubeadm reset`
+
+**Edge commands:**
+
+`kubeedge node init`
+  - What is it? 
+    * This command will be responsible to install pre-requisites KubeEdge edge component (edge_core)
```

Comment:
pre-requisites KubeEdge edge component -> pre-requisites needed to install KubeEdge edge component